### ISSUE TITLE
cant load pre-train models

### ISSUE THREAD

**Issue by AishuaiYao** [tensorflow/models#8695]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [N ] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [Y ] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [Y ] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using
https://github.com/tensorflow/models/tree/master/research/deeplab

## 2. Describe the bug

Hello, thank you for your work. I am using deeplabv3-mobilenet for finetune, but there is a problem when I load the model. The phenomenon is that the program does not report an error, but I am sure that the parameters of the model are not loaded, causing me to never get the ideal results.

## 3. Steps to reproduce

run train.py

## 4. Expected behavior

Successfully loaded model parameters

## 6. System information

No error, but there is a problem

**Comment by AishuaiYao** [comment]
I debugged the program for importing the model, but the program is too complicated. There are many details that I can't see. It feels like the model parameters are imported. However, from the point of view of the loss decline, it starts to decline from 3, 4, 5 etc. It is obviously that training is restarted, and it is not based on the pre-trained model. In addition, eval.py and vis.py are used for evaluation and visualization. The uncompressed pb file can be used normally, and the ckpt file will be obviously confusing.

**Comment by AishuaiYao** [comment]
I’m training a binary classification network. According to common sense, it should be easy to produce ideal results, but with this code, there have been problems, which have troubled me for a long time. Recently I finally found that it may be a problem with the pre-trained model. Can someone help me? I really want to solve the problem

**Comment by Beerwalker** [comment]
It seems like commit 0ad4922f5617a0bdede56fa9d5cccf26bbdc294c removed hparams from model_main.py and they are no longer passed.
Decision to load pretrained weights is made here: https://github.com/tensorflow/models/blob/950ebc455eda5bb55c4fb60712753fd8a3c6cf92/research/object_detection/model_lib.py#L446-L447
And since hparams are None then load_pretrained is always False and nothing happens.
Setting load_pretrained = True fixed it for me, however this might not be the correct way to solve this.

**Comment by swapnil3597** [comment]
> It seems like commit [0ad4922](https://github.com/tensorflow/models/commit/0ad4922f5617a0bdede56fa9d5cccf26bbdc294c) removed hparams from model_main.py and they are no longer passed.
> Decision to load pretrained weights is made here:
> 
> https://github.com/tensorflow/models/blob/950ebc455eda5bb55c4fb60712753fd8a3c6cf92/research/object_detection/model_lib.py#L446-L447
> 
> 
> And since hparams are None then load_pretrained is always False and nothing happens.
> Setting load_pretrained = True fixed it for me, however this might not be the correct way to solve this.

Worked for me. :)

**Comment by xdhmoore** [comment]
Would it not fix the code to change the end of the mentioned line from `False` to `True`?
```diff
diff --git a/research/object_detection/model_lib.py b/research/object_detection/model_lib.py
index ac2dff15..42b65cbc 100644
--- a/research/object_detection/model_lib.py
+++ b/research/object_detection/model_lib.py
@@ -478,7 +478,7 @@ def create_model_fn(detection_model_fn, configs, hparams=None, use_tpu=False,
             features[fields.InputDataFields.true_image_shape]))

     if mode == tf.estimator.ModeKeys.TRAIN:
-      load_pretrained = hparams.load_pretrained if hparams else False
+      load_pretrained = hparams.load_pretrained if hparams else True
       if train_config.fine_tune_checkpoint and load_pretrained:
         if not train_config.fine_tune_checkpoint_type:
           # train_config.from_detection_checkpoint field is deprecated. For
```

This would allow one to pass in `hparams` if needed, but if `hparams` was not passed in would use only the value of `train_config.fine_tune_checkpoint`.

Just finished running a job with this change and it seems to have loaded the checkpoint where it wasn't before.
